From fbbb55cb30029c4be6c104dd97be1e9d42c5dfec Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Fri, 16 Aug 2019 17:41:30 +0800
Subject: [PATCH] OP-TEE: fix clang warning

security/optee_linuxdriver/core/tee_session.c:683:22: error: address of array 'cmd_io->op->params' will always evaluate to 'true'
      [-Werror,-Wpointer-bool-conversion]
        BUG_ON(!cmd_io->op->params);
               ~~~~~~~~~~~~~^~~~~~
/security/optee_linuxdriver/core/tee_kernel_api.c:85:18: error: address of array 'context->devname' will always evaluate to 'true'
      [-Werror,-Wpointer-bool-conversion]
                                 && context->devname) ? context->devname : "");
                                 ~~ ~~~~~~~~~^~~~~~~

Change-Id: I6fa7e26fa620b48a0a255393b4279ff1cf1700c3
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 security/optee_linuxdriver/core/tee_kernel_api.c | 3 +--
 security/optee_linuxdriver/core/tee_session.c    | 1 -
 2 files changed, 1 insertion(+), 3 deletions(-)

diff --git a/security/optee_linuxdriver/core/tee_kernel_api.c b/security/optee_linuxdriver/core/tee_kernel_api.c
index 3b4380faf3b1..41568c8e9ec0 100644
--- a/security/optee_linuxdriver/core/tee_kernel_api.c
+++ b/security/optee_linuxdriver/core/tee_kernel_api.c
@@ -81,8 +81,7 @@ void TEEC_FinalizeContext(TEEC_Context *context)
 {
 	if (!context || !context->fd) {
 		pr_err("%s - can't release context %p:[%s]\n", __func__,
-		       context, (context
-				 && context->devname) ? context->devname : "");
+		       context, context ? context->devname : "");
 		return;
 	}
 	/* TODO fixme will not work on 64-bit platform */
diff --git a/security/optee_linuxdriver/core/tee_session.c b/security/optee_linuxdriver/core/tee_session.c
index 501e89f5c276..32ffb07a131d 100644
--- a/security/optee_linuxdriver/core/tee_session.c
+++ b/security/optee_linuxdriver/core/tee_session.c
@@ -680,7 +680,6 @@ static void _update_client_tee_cmd(struct tee_session *sess,
 
 	BUG_ON(!cmd_io);
 	BUG_ON(!cmd_io->op);
-	BUG_ON(!cmd_io->op->params);
 	BUG_ON(!cmd);
 	BUG_ON(!sess->ctx);
 	ctx = sess->ctx;
-- 
2.35.3

